### ISSUE TITLE
First segment is set to 0, same as padding tokens

### ISSUE THREAD

**Issue by mikkelfo** [FGA-DIKU/EHR#168]
In the old code we do `factorize + 1`
https://github.com/mikkelfo/CORE-BEHRT/blob/7cf32d8dcec801a5d6bfdad055ae79aa36a401cf/ehr2vec/data/creators.py#L57

The new code has dropped the + 1 at some point, which makes the `segment=0` for padding tokens equal to the first segments
https://github.com/FGA-DIKU/EHR/blob/b51021e1a41cc56b420c7bee8e6d1e13eacac173/corebehrt/functional/features/normalize.py#L39

**Comment by mikkelfo** [comment]
When tackling this issue, one should set `padding_idx` in `EhrEmbeddings` as well

**Comment by mikkelfo** [comment]
We use the `padding_idx` in our `nn.Embedding`, but can't do it for `segment` due to this
https://github.com/FGA-DIKU/EHR/blob/1543aa668dd66eda65649abd1641583eb7f511be/corebehrt/modules/model/embeddings.py#L52-L55

padding_idx ([int](https://docs.python.org/3/library/functions.html#int), optional) – If specified, the entries at padding_idx do not contribute to the gradient; therefore, the embedding vector at padding_idx is not updated during training, i.e. it remains as a fixed “pad”. For a newly constructed Embedding, the embedding vector at padding_idx will default to all zeros, but can be updated to another value to be used as the padding vector.

**Comment by kirilklein** [comment]
Also I guess now it would be a breaking change, since we then cannot run with existing tokenized features.

**Comment by mikkelfo** [comment]
Yeah, I think that was the issue, that we'd have to re-run data processing, which I didn't want to do at the time - but I'm not 100% sure if that was the reason.
